### PR TITLE
feat: WebSocket notifier

### PR DESCRIPTION
**Pull request by troyronda**
- adds a WebSocket notification endpoint
- refactors the webhook package into webnotification

TODO: BDD tests

Closes: #471

Signed-off-by: Troy Ronda <troy@troyronda.com>

### CODE REVIEW

File: test/bdd/agent/agent_controller_steps.go [hunk 1]
```diff
@@ -140,15 +140,11 @@ func (a *ControllerSteps) healthCheck(url string) error {
 
 		return nil
 	} else if strings.HasPrefix(url, "ws") {
-		_, resp, err := websocket.Dial(goctx.Background(), url, nil)
+		_, _, err := websocket.Dial(goctx.Background(), url, nil) //nolint:bodyclose
 		if err != nil {
 			return err
 		}
 
-		if err := resp.Body.Close(); err != nil {
```
Comment (troyronda) [inline test/bdd/agent/agent_controller_steps.go]: In latest WebSocket package, resp.Body isn't set after Dial.

File: pkg/didcomm/transport/ws/inbound_test.go [hunk 2]
```diff
@@ -152,7 +152,7 @@ func TestInboundDataProcessing(t *testing.T) {
 		// create ws client
 		client, _ := websocketClient(t, port)
 
-		err = client.Close(websocket.StatusAbnormalClosure, "abnormal closure")
```
Comment (troyronda) [inline pkg/didcomm/transport/ws/inbound_test.go]: In latest WebSocket library, you are not allowed to set this status.

File: pkg/controller/webnotifier/webnotifier.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	return &n
+}
+
+// Notify sends the given message to all of the subscribers.
+// If multiple errors are encountered, then the first one is returned.
+func (n *WebNotifier) Notify(topic string, message []byte) error {
+	var allErrs error
+
+	for _, notifier := range n.notifiers {
```
Comment (llorllale) [inline pkg/controller/webnotifier/webnotifier.go]: at some point (multitenancy) we'll need to direct notifications to their subscribers

Comment (troyronda) [reply pkg/controller/webnotifier/webnotifier.go]: yup - just following the existing code in this PR.

**Pull request merged**